CMakeLists.txt

```diff
@@ -922,6 +931,38 @@ elseif(WIN32)
   set(CPACK_BINARY_NSIS OFF CACHE BOOL "Enable to build NSIS packages")
 endif()
 
+
+if(APPLE)
+
+  if(CPACK_CODESIGNING_DEVELOPPER_ID_APPLICATION)
+    # Codesign inner binaries and libraries, in the CPack staging area for the current project & component
+    # Define some required variables for the script in the scope of the install(SCRIPT) first
+    set(CLI_FILES_TO_SIGN
+      lib/libopenstudiolib.dylib
+      lib/libpythonengine.so
+      lib/librubyengine.so
+      bin/install_utility
+    )
+    install(CODE "set(FILES_TO_SIGN \"${CLI_FILES_TO_SIGN}\")" COMPONENT "CLI")
+    install(CODE "set(CPACK_CODESIGNING_DEVELOPPER_ID_APPLICATION \"${CPACK_CODESIGNING_DEVELOPPER_ID_APPLICATION}\")" COMPONENT CLI)
+    install(CODE "set(CPACK_CODESIGNING_MACOS_IDENTIFIER \"${CPACK_CODESIGNING_MACOS_IDENTIFIER}\")" COMPONENT CLI)
+    install(SCRIPT "${CMAKE_CURRENT_LIST_DIR}/CMake/install_codesign_script.cmake" COMPONENT CLI)
```

Comment:
Does this signing happen at install time on the users computer?  Does this leak our certificate information into the installer?

Reply:
It happens in the cpack staging area. So build/_Cpack_Packages/darwin/<tgz or ifw>

So no, no leaking